src/goto-analyzer/unreachable_instructions.cpp

```diff
@@ -243,22 +250,26 @@ bool static_unreachable_instructions(
   return false;
 }
 
+static std::string line_or_zero(const source_locationt &source_location)
+{
+  const irep_idt &line = source_location.get_line();
 
+  if(line.empty())
+    return "0";
+  else
+    return id2string(line);
+}
 
 static void json_output_function(
   const irep_idt &function,
   const source_locationt &first_location,
   const source_locationt &last_location,
   json_arrayt &dest)
 {
-  json_objectt entry{
-    {"function", json_stringt(function)},
-    {"file name",
-     json_stringt(concat_dir_file(
-       id2string(first_location.get_working_directory()),
-       id2string(first_location.get_file())))},
-    {"first line", json_numbert(id2string(first_location.get_line()))},
-    {"last line", json_numbert(id2string(last_location.get_line()))}};
+  json_objectt entry{{"function", json_stringt(function)},
```

Comment:
It would be preferable to omit the property if we don't have information instead of using an empty string and 0.

Reply:
Good call, that's now done.